llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -1519,17 +1521,21 @@ bool LoopVectorizationLegality::canVectorize(bool UseVPlanNativePath) {
       return false;
   }
 
-  LLVM_DEBUG(dbgs() << "LV: We can vectorize this loop"
-                    << (LAI->getRuntimePointerChecking()->Need
-                            ? " (with a runtime bound check)"
-                            : "")
-                    << "!\n");
-
   unsigned SCEVThreshold = VectorizeSCEVCheckThreshold;
   if (Hints->getForce() == LoopVectorizeHints::FK_Enabled)
     SCEVThreshold = PragmaVectorizeSCEVCheckThreshold;
 
+  if (Result) {
```

Comment:
also keep where the message is emitted at the original place, so `SCEVThreshold`'s definition is closer to use?

Reply:
Oh yes, there is no reason to change. Thanks! Hopefully, looks good now :)